docs/guide-pl/input-validation.md

```diff
@@ -0,0 +1,649 @@
+
+```php
+<?php $form = yii\widgets\ActiveForm::begin(); ?>
+    <?= $form->field($model, 'username') ?>
+    <?= $form->field($model, 'password')->passwordInput() ?>
+    <?= Html::submitButton('Login') ?>
+<?php yii\widgets\ActiveForm::end(); ?>
+```
+
+"Za kulisami", widżet [[yii\widgets\ActiveForm|ActiveForm]] odczyta wszystkie zasady walidacji zadeklarowane w modelu i wygeneruje odpowiedni kod JavaScript 
+dla walidatorów wspierających walidację po stronie klienta. Kiedy użytkownik zmieni wartość w polu lub spróbuje wysłać formularz, zostanie wywołana walidacja po stronie klienta.
+
+Jeśli chcesz wyłączyć kompletnie walidację po stronie klienta, możesz ustawić właściwość [[yii\widgets\ActiveForm::enableClientValidation|enableClientValidation]] na `false`.
+Możesz również wyłączyć ten rodzaj walidacji dla konkretnego pola, przez ustawienie jego właściwości 
+[[yii\widgets\ActiveField::enableClientValidation|enableClientValidation]] na `false`. Jeśli właściwość `enableClientValidation` zostanie skonfigurowana na poziomie pola formularza i w 
+samym formularzu jednocześnie, pierwszeństwo będzie miała opcja określona w formularzu.
+
+
+### Implementacja walidacji po stronie klienta <span id="implementing-client-side-validation"></span>
+
+Aby utworzyć walidator wspierający walidację po stronie klienta, powinieneś zaimplementować metodę [[yii\validators\Validator::clientValidateAttribute()|clientValidateAttribute()]],
+która zwraca kod JavaScript, odpowiedzialny za przeprowadzenie walidacji. W kodzie JavaScript możesz użyć następujących predefiniowanych zmiennych:
+
+- `attribute`: nazwa atrybutu podlegającego walidacji.
+- `value`: wartość atrybutu podlegająca walidacji.
+- `messages`: tablica używana do przechowywania wiadomości błędów dla danego atrybutu. 
+- `deferred`: tablica, do której można dodać zakolejkowane obiekty (wyjaśnione w późniejszej podsekcji).
+
+W poniższym przykładzie, tworzymy walidator `StatusValidator`, który sprawdza, czy wartość danego atrybutu jest wartością znajdującą się w zbiorze statusów w bazie danych.
+Walidator wspiera obydwa typy walidacji: po stronie klienta oraz serwerową.
+
+```php
+namespace app\components;
+
+use yii\validators\Validator;
+use app\models\Status;
+
+class StatusValidator extends Validator
+{
+    public function init()
+    {
+        parent::init();
+        $this->message = 'Niepoprawna wartość pola status.';
+    }
+
+    public function validateAttribute($model, $attribute)
+    {
+        $value = $model->$attribute;
+        if (!Status::find()->where(['id' => $value])->exists()) {
+            $model->addError($attribute, $this->message);
+        }
+    }
+
+    public function clientValidateAttribute($model, $attribute, $view)
+    {
+        $statuses = json_encode(Status::find()->select('id')->asArray()->column());
+        $message = json_encode($this->message, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+        return <<<JS
+if ($.inArray(value, $statuses) > -1) {
```

Comment:
Recently there was a PR that fixed this expression: https://github.com/yiisoft/yii2/commit/83d72cddc02da030a384e87295721ea6a2ce2937